src/masonite/container/container.py

```diff
@@ -181,7 +181,7 @@ def resolve(self, obj, *resolving_arguments):
                 raise ContainerError(str(e))
         else:
             for _, value in self.get_parameters(obj):
-                if value.annotation in (str, int, dict, list, tuple):
+                if type(value.annotation) in (str, int, dict, list, tuple):
```

Comment:
This entire problem was due to this line right here lol. This line i guess never worked. value.annotation was something like `"app: Application"` (a string). So that value was never in this list of types lol. We had to do the type(value.annotation) in order to check the type against the list of types. Changing this fixed the issue.

We now get an expected error like "__init__() missing required parameter app"

Reply:
Good news !